### PR TITLE
Fixes [#241] Prevent Webpack compilers from running on every page. 

### PR DESCRIPTION
**Pull request by petercr**
Colby, here's the PR with my fix for Issue #241 to stop the Webpack compiler from running on every page compile in development.

While clicking through my local build these changes seemed to cut down on the amount of time needed to rebuild the pages, and the number of plugin builds as well. I think that was the real time saver.  
It was noticeably faster on my machine, but maybe double check with yours and see if you get the same results.

Basically, I just declared a `hasRun` variable to hold the compile state for the plugins. 
I doubt that this would have any effect in production, since that's a static build. 

Let me know if you have any questions, or want me to change anything 👍🏻

Thanks

### CODE REVIEW

**Comment by colbyfayock** [comment]
oh i just realized i think the issue with the failing static compat checks is your PR i dont think can see the env var set. not sure how i can fix that but for now i od see its working

thanks for getting this together!

**Comment by colbyfayock** [comment]
@all-contributors please add @petercr for code
